internal/controllers/machineset/machineset_controller.go

```diff
@@ -966,6 +1003,156 @@ func (r *Reconciler) getMachineNode(ctx context.Context, cluster *clusterv1.Clus
 	return node, nil
 }
 
+func (r *Reconciler) runPreFlightChecks(ctx context.Context, cluster *clusterv1.Cluster, ms *clusterv1.MachineSet) (ctrl.Result, error) {
```

Comment:
WDYT about moving this func and funcs that we use in this func to machineset_preflight.go?

Reply:
Moved the functions (and the tests) to separate file.